### PR TITLE
Implement verifier alliance tests

### PR DESCRIPTION
**Pull request by marcocastignoli**
<sub>View in Huly <a href="https://app.huly.io/guest/w-kaanuzdogan-sourcify-65d30de0-75121c5c6c-fd615e?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NWU2MTc3MjhiNmVjZDA0ZDYyZGIwNzciLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6Incta2FhbnV6ZG9nYW4tc291cmNpZnktNjVkMzBkZTAtNzUxMjFjNWM2Yy1mZDYxNWUiLCJwcm9kdWN0SWQiOiIifQ.9CiXsXfn15_yp6tpkJyztx_cI0kmlnIlrbmdM-jsfR4">HI-1247</a></sub>

### CODE REVIEW

File: packages/lib-sourcify/src/lib/verification.ts [hunk 1]
```diff
@@ -455,7 +467,14 @@ export async function matchWithCreationTx(
...
   recompiledCreationBytecode = replaced;
-  match.creationTransformationValues.libraries = libraryMap;
+  if (Object.keys(libraryMap).length > 0) {
+    match.creationTransformationValues.libraries = Object.keys(
+      libraryMap
+    ).reduce((libMap: any, lib) => {
+      libMap[lib] = `0x${libraryMap[lib]}`;
+      return libMap;
+    }, {});
+  }
```
Comment (kuzdogan) [inline packages/lib-sourcify/src/lib/verification.ts:477]: Same comment as above for the runtime match. Reduce overkill?

File: services/server/src/server/services/storageServices/AbstractDatabaseService.ts [hunk 2]
```diff
@@ -98,6 +98,7 @@ export default abstract class AbstractDatabaseService {
       userdoc: compilerOutput?.userdoc || {},
       devdoc: compilerOutput?.devdoc || {},
       storageLayout: compilerOutput?.storageLayout || {},
+      sources: recompiledContract.compilerOutput?.sources || {},
```
Comment (kuzdogan) [inline services/server/src/server/services/storageServices/AbstractDatabaseService.ts:101]: Where is this used?

**Review by kuzdogan — COMMENTED** [review]
Couple remarks to be cleared. Otherwise looking good!

File: packages/lib-sourcify/src/lib/CheckedContract.ts [hunk 3]
```diff
@@ -647,6 +651,7 @@ function createJsonInputFromMetadata(
     'evm.bytecode.object',
     'evm.bytecode.sourceMap',
     'evm.bytecode.linkReferences',
+    'evm.bytecode.generatedSources',
```
[the remark this answers is not in the document]
Comment (marcocastignoli) [reply packages/lib-sourcify/src/lib/CheckedContract.ts:654]: It's used to save the sources into the compilation artifacts here:

https://github.com/ethereum/sourcify/blob/a52932fa10bbdf123c99a9f05688670658d54c93/services/server/src/server/services/utils/database-util.ts#L43

File: services/server/src/server/services/storageServices/AbstractDatabaseService.ts (hunk 2 above)
Comment (marcocastignoli) [reply services/server/src/server/services/storageServices/AbstractDatabaseService.ts:101]: It's an information stored in the database. See the comment above. It's part of blockscout's tests.

**Pull request merged**